extensions/objects/templates/FilepathEditor.js

```diff
@@ -73,27 +162,321 @@ oppia.directive('filepathEditor', [
           }
         });
 
+        $scope.resetFilePathEditor = function() {
+          $scope.data = {
+            mode: MODE_EMPTY,
+            metadata: {}
+          };
+          $scope.imageResizeRatio = 1;
+        };
+
         $scope.validate = function(data) {
           return data.mode === MODE_SAVED &&
                  data.metadata.savedUrl &&
                  data.metadata.savedUrl.length > 0;
         };
 
+        $scope.isUserCropping = function() {
+          var dimensions = $scope.calculateTargetImageDimensions();
+          var cropWidth = $scope.croppingArea.x2 - $scope.croppingArea.x1;
+          var cropHeight = $scope.croppingArea.y2 - $scope.croppingArea.y1;
+          return cropWidth < dimensions.width || cropHeight < dimensions.height;
+        };
+
+        $scope.mouseMoveWhileDraggingCroppingArea = function(x, y) {
+          var xDown = $scope.mouseLastDownCoordinates.x;
+          var yDown = $scope.mouseLastDownCoordinates.y;
+          var x1WhenDown = $scope.croppingAreaXWhenLastDown;
```

Comment:
I renamed it to "lastMouseDownEventCoordinates". The naming was from the beginning referring to the MouseDown event, so that's why I used it. Let me know what you think.

I see what you're saying about Point. Maybe if it was going to be used globally across Oppia, but if it is for just this component, I'd find it overkill.